### PR TITLE
DFP-2759 Removed secret information from logs

### PR DESCRIPTION
**Pull request by anuragverma099**

### CODE REVIEW

File: helix_fhir_client_sdk/fhir_client.py [hunk 1]
```diff
@@ -1620,6 +1623,21 @@ async def refresh_token(
...
 
+    @staticmethod
+    def get_variables_to_log(vars_dict: Dict[str, Any]) -> Dict[str, Any]:
+        """
+        Method to return the variables which we need to log
+        :param vars_dict: (dict) dictionary of variables names with their values
+        """
+        variables_to_log = {}
+        for key, value in vars_dict.items():
+            if not isinstance(value, type(threading.Lock)):
```
Comment (shubhamgoeljtg) [inline helix_fhir_client_sdk/fhir_client.py:1634]: Is there any case where this type of value is coming? Please mention here.

File: helix_fhir_client_sdk/fhir_client.py [hunk 2]
```diff
@@ -1620,6 +1623,21 @@ async def refresh_token(
...
+                variables_to_log[key] = value
+        variables_to_log.pop("_access_token")
+        variables_to_log.pop("_login_token")
+        variables_to_log.pop("_client_id")
+        return variables_to_log
```
Comment (shubhamgoeljtg) [inline helix_fhir_client_sdk/fhir_client.py:1639]: Are we getting only 3 types of secret logs in every case?

File: helix_fhir_client_sdk/fhir_client.py (hunk 1 above)
Comment (anuragverma099) [reply helix_fhir_client_sdk/fhir_client.py:1634]: there are some variables which contains objects. On these objects we are having lock due to which we were not able to add those into our variables dict which we need to log.

File: helix_fhir_client_sdk/fhir_client.py (hunk 2 above)
Comment (anuragverma099) [reply helix_fhir_client_sdk/fhir_client.py:1639]: yes

**Review by shubhamgoeljtg — APPROVED** [review]

**Comment by cdukebwell** [comment]
Is there any way to create a test to verify this is producing the expected logs?

**Review by shubhamgoeljtg — APPROVED** [review]

**Pull request merged**